recipes/newrelic/apm/php/debian.yml

```diff
@@ -465,10 +465,20 @@ install:
               # Get the PHP INI Directory associated with this specific NR INI file
               #
               php_ini_dir=$(echo $ini | sed -n 's/\(.*\)\/conf.d/\1/p')
-              if [ -z "${php_ini_dir}" ]; then
-                php_ini_dir=$ini
+              if [[ -z "${php_ini_dir}" ]]; then
+                if [[ -z "${CLI_DIRS}" ]]; then
+                    if [[ -d "${ini}/../apache2" ]]; then
+                      php_ini_dir="${ini}/../apache2"
+                    fi
+                    if [[ -d "${ini}/../fpm" ]]; then
+                      php_ini_dir="${ini}/../fpm"
+                    fi
+                elif [[ -d "${ini}/../cli" ]]; then
+                  php_ini_dir="${ini}/../cli"
+                else
+                  php_ini_dir=$ini
+                fi
```

Comment:
**Changes Explained:**

As stated by the PHP Agent team, with the latest PHP Agent v11.3.0.16 released around two weeks ago, they've introduced a change, which would affect the functioning of the PHP tarball installer. 

With the debian (package) installer using the `etc/php/7.x/mods-available` directory (by design) to configure `newrelic.ini`, customers tending to upgrade using the tarball installer would sometimes run into issues, since the installer would place a duplicate `newrelic.ini` in `etc/php/7.x` `apache2` (or) `fpm` (or) `cli`, leading to duplicates of the `newrelic.ini` file. To solve this problem, the solution devised by the agent team was to allow for "skipping" placing a `newrelic.ini` file elsewhere, if a `newrelic.ini` is already found to have been present in the `mods-available` directory by the tarball installer.

This affected the recipe owing to the following reason. 

The source of `${ini}` here is a collection of probable directories where the `php.ini` file could be present, originating from `pi_inidir_dso` and/or `pi_inidir_cli` originating from [this place in the agent](https://github.com/newrelic/newrelic-php-agent/blob/f30642fd4f51514d4ba507ca643cb8767327b922/agent/newrelic-install.sh#L1360-L1394), which the CLI scrapes.

This list earlier contained a bunch of directories such as `apache2` or `fpm` or `cli`, which would contain a `conf.d` and `php.ini`; however, in order to allow for the tarball installer to check for the presence of a `newrelic.ini` in `mods-available`, this has also been in the code linked above, to override these paths when present. This wouldn't cause an issue with the agent, since the intended use is to skip populating a `newrelic.ini` in any of the aforementioned directories, if there is a `mods-available` with a `newrelic.ini`. 

However, in the case of this recipe, since we rely on the `${ini}` picked later in the code to set the `target_ini_dir` where the `newrelic.ini` would need to be moved, the value of `${ini}`, which would ideally need to be one of the aforementioned directories and then allow traversing to `mods-available` to overwrite the `newrelic.ini` file there, is now `mods-available` already, which is why it is not able to traverse out of `mods-available` to `mods-available`; which is exactly what the error upon guided install currently states.

To avoid this problem, two things needed to be done - 

- firstly, if `${ini}` is bound to be `mods-available` whenever a `newrelic.ini` is present in `mods-available`, reverse engineer the conditions to fix the effective `${ini}` we shall use to move the `newrelic.ini` to, using `php_ini_dir`. In other words, conditionally assign `${ini}` to `php_ini_dir` based on the path found at `../` of `${ini}`, which would in turn, go back to the expected `apache2` or `fpm` or `cli` based on the conditions written. If none of the conditions are satisfied, `php_ini_dir` takes `${ini}`, which is also likely to happen when `${ini}` is not `mods-available`, so either way, this conditional logic prevents `php_ini_dir` from being assigned to the wrong directory, especially since this is used later in the code.

- the condition to configure `target_ini_dir` was earlier configured based on `${ini}` which may now be made `php_ini_dir` (with one lesser folder to traverse out of, `../` instead of `../../`) since `php_ini_dir` **_now takes the "effective"_** `${ini}` and **_not the actual_** `${ini}` directly, making sure that the traversal would never fail, which would prevent the error we're seeing now